rosserial_python/src/rosserial_python/SerialClient.py

```diff
@@ -56,6 +65,15 @@
 import struct
 import time
 
+import sys
+if sys.version_info[0] >= 3:
+    def chr(a):
+        return bytes([a])
+    def ord(a):
+        if isinstance(a, int):
+            return a
+        else:
+            return int.from_bytes(a, 'little')
```

Comment:
Rather than defining these functions for Py3, I wonder if we could instead just adapt usage of them to use `struct` instead? https://docs.python.org/2/library/struct.html

Reply:
Hmm. The solution in the PR is a shim for Python 2 functions just to MAKE IT WORK 😈 Actually I think the code needs a complete rewrite, too, however that would be a huge amount of work :) I am pretty happy to do the hard work, but we can merge this first to enable py3 usage.